disttask/framework/dispatcher/dispatcher.go

```diff
@@ -206,12 +223,75 @@ func (d *dispatcher) onRunning() error {
 		logutil.Logger(d.logCtx).Info("previous stage finished, generate dist plan", zap.Int64("stage", d.task.Step))
 		return d.onNextStage()
 	}
+	// Check if any node are down.
+	handle := GetTaskFlowHandle(d.task.Type)
+	if err := d.replaceDeadNodesIfAny(handle); err != nil {
+		return err
+	}
 	// Wait all subtasks in this stage finished.
-	GetTaskFlowHandle(d.task.Type).OnTicker(d.ctx, d.task)
+	handle.OnTicker(d.ctx, d.task)
 	logutil.Logger(d.logCtx).Debug("on running state, this task keeps current state", zap.String("state", d.task.State))
 	return nil
 }
 
+func (d *dispatcher) replaceDeadNodesIfAny(handle TaskFlowHandle) error {
+	if len(d.taskNodes) == 0 {
```

Comment:
When the subtasks enter step 2, `d.taskNodes` still tracks the subtask in step 1.

If there is a new node added before dispatching step 2 subtasks, it may not be recorded in `d.taskNodes`.